plumbing/format/index/decoder.go

```diff
@@ -24,8 +24,8 @@ var (
 	// ErrInvalidChecksum is returned by Decode if the SHA1 hash mismatch with
 	// the read content
 	ErrInvalidChecksum = errors.New("invalid checksum")
-
-	errUnknownExtension = errors.New("unknown extension")
+	// ErrUnknownMandatoryExtension is returned when an index extension is encountered that is considered mandatory
+	ErrUnknownMandatoryExtension = errors.New("unknown mandatory extension")
```

Comment:
As this is an existing error, and aligns with [upstream](https://github.com/git/git/blob/19981daefd7c147444462739375462b49412ce33/setup.c#L745), we probably should not change it on a patch version.

```suggestion
	ErrUnknownExtension = errors.New("unknown extension")
```